include/metrics/Timer.h

```diff
@@ -53,34 +44,28 @@ class Timer
 
     // noexcept moveable
     explicit Timer(Timer&& other) noexcept
-    : _timer( other._timer )
-    , _start( std::move(other._start) )
+    : _timer(std::move(other._timer))
     {
-        other._timer = nullptr;
     }
 
     Timer& operator=(Timer&& other) noexcept {
         if (&other != this) {
-            _timer = other._timer;
-            other._timer = nullptr;
-            _start = std::move( other._start );
+            _timer = std::move( other._timer );
         }
         return *this;
     }
 
-    void reset(metric_type* timer = nullptr,
+    void reset(metric_type* metric = nullptr,
         prometheus::Counter* start_counter = nullptr)
     {
-        timespec now;
-        clock_gettime(CLOCK_MONOTONIC, &now);
-        if (_timer) {
-            duration_type dur =
-                std::chrono::nanoseconds(now.tv_nsec - _start.tv_nsec) +
-                std::chrono::seconds(now.tv_sec - _start.tv_sec);
-            _timer->Observe(dur.count());
+        if (metric) {
```

Comment:
so what this does here is create the ScopeTimer, and creates a function which on the ScopeTimer _timer destructor pushes the value out and into metric?

Reply:
yes, exactly.  In other words, creates an object that when it is destructed will automatically record the duration of its existence to `metric`.

Reply:
nice, makes the metrics timer very clean